client/asset/eth/eth.go

```diff
@@ -331,11 +331,17 @@ func (eth *ExchangeWallet) Connect(ctx context.Context) (*sync.WaitGroup, error)
 	return &wg, nil
 }
 
-// OwnsAddress indicates if an address belongs to the wallet.
+// OwnsAddress indicates if an address belongs to the wallet. The address need
+// not be a EIP55-compliant formatted address. It may or may not have a 0x
+// prefix, and case is not important.
 //
 // In Ethereum, an address is an account.
 func (eth *ExchangeWallet) OwnsAddress(address string) (bool, error) {
-	return strings.ToLower(eth.acct.Address.String()) == strings.ToLower(address), nil
+	if !common.IsHexAddress(address) {
+		return false, errors.New("invalid address")
+	}
+	addr := common.HexToAddress(address)
```

Comment:
`common.NewMixedCaseAddress` would work fine for this too, but it has a little extra overhead we don't need